samples/nrf9160/mqtt_simple/src/main.c

```diff
@@ -395,7 +459,10 @@ void main(void)
 
 	modem_configure();
 
-	client_init(&client);
+	err = client_init(&client);
+	if (err != 0) {
+		printk("ERROR: client_init %d\n", err);
```

Comment:
It looks like an error here could cause the subsequent functions to fail (depending on peer verification settings) so it should be considered to return.

Slightly besides the scope of this PR: I think it should also be considered to let `broker_init()` return an error and propagate it from `client_init()`. Error handling in `broker_init()` looks incomplete, and failure in that function would cause the connection attempt to fail.

Reply:
Let's add this while we are at it.